service/src/test/java/uk/gov/hmcts/reform/fpl/service/ConfirmApplicationReviewedServiceTest.java

```diff
@@ -0,0 +1,150 @@
+        element(AdditionalApplicationsBundle.builder()
+            .uploadedDateTime("2022 Dec 30")
+            .author("TESTING2")
+            .otherApplicationsBundle(OtherApplicationsBundle.builder()
+                .applicationType(OtherApplicationType.C1_TERMINATION_OF_APPOINTMENT_OF_A_GUARDIAN)
+                .uploadedDateTime("2022 Dec 30").build())
+            .build());
+
+    @Test
+    void shouldInitEventFieldWithListOfBundlesToBeReviewed() {
+        CaseData caseData = CaseData.builder()
+            .additionalApplicationsBundle(List.of(REVIEWED_BUNDLE, NEW_BUNDLE_1, NEW_BUNDLE_2))
+            .build();
+
+        Map<String, Object> resultMap = confirmApplicationReviewedService.initEventField(caseData);
+
+        Map<String, Object> expectedMap = Map.of(
+            "hasApplicationToBeReviewed", YES,
+            "confirmApplicationReviewedList",
+            asDynamicList(List.of(NEW_BUNDLE_1, NEW_BUNDLE_2), AdditionalApplicationsBundle::toLabel)
+        );
+
+        assertThat(resultMap).isEqualTo(expectedMap);
+    }
+
+    @Test
+    void shouldInitEventFieldWithOutBundlesToBeReviewed() {
+        CaseData caseData = CaseData.builder()
+            .additionalApplicationsBundle(List.of(REVIEWED_BUNDLE))
+            .build();
+
+        Map<String, Object> resultMap = confirmApplicationReviewedService.initEventField(caseData);
+
+        Map<String, Object> expectedMap = Map.of("hasApplicationToBeReviewed", NO);
+
+        assertThat(resultMap).isEqualTo(expectedMap);
+    }
+
+    @Test
+    void shouldGetSelectedApplicationsToBeReviewed() {
+        CaseData caseData = CaseData.builder()
+            .additionalApplicationsBundle(List.of(REVIEWED_BUNDLE, NEW_BUNDLE_1, NEW_BUNDLE_2))
+            .confirmApplicationReviewedEventData(ConfirmApplicationReviewedEventData.builder()
+                .confirmApplicationReviewedList(asDynamicList(List.of(NEW_BUNDLE_1, NEW_BUNDLE_2),
+                    NEW_BUNDLE_1.getId(), AdditionalApplicationsBundle::toLabel))
+                .build())
+            .build();
+
+        Element<AdditionalApplicationsBundle> result =
+            confirmApplicationReviewedService.getSelectedApplicationsToBeReviewed(caseData);
+
+        assertThat(result).isEqualTo(NEW_BUNDLE_1);
+    }
+
+    @Test
+    void shouldThrowExceptionWhenSelectedBundleNotFound() {
+        CaseData caseData = CaseData.builder()
+            .additionalApplicationsBundle(List.of(REVIEWED_BUNDLE, NEW_BUNDLE_1, NEW_BUNDLE_2))
+            .confirmApplicationReviewedEventData(ConfirmApplicationReviewedEventData.builder()
+                .confirmApplicationReviewedList(asDynamicList(List.of(NEW_BUNDLE_1, NEW_BUNDLE_2),
+                    UUID.randomUUID(), AdditionalApplicationsBundle::toLabel))
+                .build())
+            .build();
+
+        assertThatThrownBy(() -> confirmApplicationReviewedService.getSelectedApplicationsToBeReviewed(caseData));
```

Comment:
Should test what is the exception thrown? Like
```
        assertThatThrownBy(() -> configuration.getCafcass(null))
            .isInstanceOf(NullPointerException.class)
            .hasMessage("Local authority code cannot be null");
```

Reply:
updated